src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -604,32 +604,15 @@ public IActionResult BuildTransaction([FromBody] BuildTransactionRequest request
                         Amount = output.Value,
                     });
                 }
-                
-                var result = await this.broadcasterManager.TryBroadcastAsync(transaction).ConfigureAwait(false);
-                if (result == Bitcoin.Broadcasting.Success.Yes)
-                {
-                    return this.Json(model);
-                }
 
-                if (result == Bitcoin.Broadcasting.Success.DontKnow)
+                bool broadcasted = await this.broadcasterManager.TryBroadcastAsync(transaction).ConfigureAwait(false);
+
+                if (broadcasted)
+                    return this.Json(model);
+                else
                 {
-                    // wait for propagation
-                    var waited = TimeSpan.Zero;
-                    var period = TimeSpan.FromSeconds(1);
-                    while (TimeSpan.FromSeconds(21) > waited)
-                    {
-                        // if broadcasts doesn't contain then success
-                        var transactionEntry = this.broadcasterManager.GetTransaction(transaction.GetHash());
-                        if (transactionEntry != null && transactionEntry.State == Bitcoin.Broadcasting.State.Propagated)
-                        {
-                            return this.Json(model);
-                        }
-                        await Task.Delay(period).ConfigureAwait(false);
-                        waited += period;
-                    }
+                    throw new Exception("Transaction was not validated by mempool. Can't send the transaction.");
```

Comment:
There isn't necessarily a mempool, for example when using the light wallet.

Reply:
LightWalletBroadcasterManager returns true or throws. So in case of this exception it will always be a full node.

Reply:
And here you're assuming you know exactly what's happening inside of TryBroadcastAsync. You shouldn't. Someone could come and change the LightWalletBroadcasterManager and return false (after all it returns a bool so it's perfectly allowed) and your code suddenly wouldn't make any sense.